core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+            b.hintElidedType(implicitly[FastTypeTag[T]])
+            // Ask directly to the scope instead of relying on `p`
+            val pickler = implicitly[Pickler[T]]
+            pickler.pickle(picklee.value, b)
+          })
+          builder.endEntry()
+
+        }
+
+        def unpickle(tag: String, reader: PReader) = {
+
+          val reader1 = reader.readField("value")
+          reader1.hintElidedType(implicitly[FastTypeTag[T]])
+          val tag1 = reader1.beginEntry()
+            // Ask directly to the scope instead of relying on `u`
+          val unpickler = implicitly[Unpickler[T]]
+          val result = unpickler.unpickle(tag1, reader1).asInstanceOf[T]
+          reader1.endEntry()
+
+        }
+      }
+
+}
+
+class StaticJavaRuntimeCustomPickler extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+  import ContainerPickling.containerPicklerUnpickler
+
+  test("runtime instantiated custom pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = containerPicklerUnpickler[List[Int]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
+    val format = implicitly[PickleFormat]
+    val unknownContainer = c.asInstanceOf[Container[Any]]
+    val pickled2 = runtimePickler.pickle(unknownContainer, format.createBuilder())
+    println(pickled)
+    println(pickled2)
+
+    assert(pickled === pickled2)
+
+  }
+
+  test("runtime instantiated custom unpickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val pickled = c.pickle.value
+    val blob = JSONPickle(pickled)
+    val reader = pickleFormat.createReader(blob)
+
+    val containerUnpickler = containerPicklerUnpickler[List[Int]]
+    val unpicklerClassName = containerUnpickler.getClass.getName
+    val runtimeUnpickler = RuntimeHelper.getInstance[Unpickler[Container[Any]]](unpicklerClassName)
+    
+    // Very important, assume it's elided
+    val tag = runtimeUnpickler.tag
+    assert(tag != null)
+
+    val unpickled = runtimeUnpickler.unpickle(tag.key, reader)
+    assert(c === unpickled)
+
+  }
+
+}
+
+class StaticJavaRuntimeCustomPicklerImplicitly extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+  import ContainerPickling.containerPicklerUnpicklerImplicitly
+
+  test("runtime instantiated custom pickler calling implicitly should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = containerPicklerUnpicklerImplicitly[List[Int]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
+    val format = implicitly[PickleFormat]
+    val unknownContainer = c.asInstanceOf[Container[Any]]
+    val pickled2 = runtimePickler.pickle(unknownContainer, format.createBuilder())
```

Comment:
Well, the problem is that there's no default constructor in `Pickler`/`Unpickler` and therefore it fails to execute the normal java `getInstance`.
  
Yes, as I said in my commit message the NPE does perfectly make sense (in the second test case), what I'm trying to do is to find a solution for the first and third case (in the third test case, even if we instantiate an unpickler and the fields are null, every implicitly call should return the actual objects I'm asking for, and they don't).
  
It'd fantastic if we can make this work. Aside from this, I'll look into how `AnyPickler` works more carefully. Still, I have my doubts about registering my picklers. In my previous comment, I explain you why. If I'm wrong, please let me know! :smile: But basically, i think that the registration of picklers/unpicklers should be done by scala-pickling any time a `Pickler`/`Unpickler` is generated. Note that I'm developing a library and thus I'd like to avoid forcing the final user who will eventually write the business logic of the clients and servers touch anything related to scala pickling and make this whole process more transparent.